File: Backend/src/main/java/com/example/Backend/Model/InterviewSession.java
package com.example.Backend.Model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "interview_sessions")
public class InterviewSession extends AuditModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String fromTimeInterval;
    private String toTimeInterval;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "jobSeeker_id")
    private User jobSeeker;

    @JsonBackReference
    @OneToOne
    @JoinColumn(name = "jobRequest_id")
    private JobDemande jobRequest;

    private String jobName;

    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "enterprise_id")
    private User enterprise;

}

File: Frontend/src/app/Models/InterviewSession.ts
export default class InterviewSession {

  id: number;
  jobRequestId: number;
  fromTimeInterval: string;
  toTimeInterval: string;
  jobName: string;
  enterpriseUsername: string;
  jobSeekerName: string;

  constructor(jobRequestId: number, fromTimeInterval: string, toTimeInterval: string, jobName: string, enterpriseUsername: string, jobSeekerName: string) {
    this.jobRequestId = jobRequestId;
    this.fromTimeInterval = fromTimeInterval;
    this.toTimeInterval = toTimeInterval;
    this.jobName = jobName;
    this.enterpriseUsername = enterpriseUsername;
    this.jobSeekerName = jobSeekerName;
  }

}
